internal/docs/pubsub/design.md

```diff
@@ -136,11 +126,7 @@ func main() {
 
 func receive() error {
 	ctx := context.Background()
-	client, err := acmepubsub.NewClient(ctx, "unicornvideohub")
-	if err != nil {
-		return err
-	}
-	s, err := client.OpenSubscription(ctx, "signup-minder", nil)
+	s, err := acmepubsub.OpenSubscription(ctx, "unicornvideohub", "signup-minder", nil)
```

Comment:
It's unlikely that there wouldn't be some sort of client or connection that is shared for all topics and subscriptions.

Reply:
It appears we have a lack of consensus about this. I'll set up a meeting to resolve it.

Reply:
I believe there to be consensus; I agree with @jba's statement. What is in this PR does not currently match what I proposed in #682. This PR proposes:

```go
func OpenTopic(ctx context.Context, projectID, topicName string) (*pubsub.Topic, error)
```

whereas I proposed something like:

```go
func OpenTopic(ctx context.Context, conn *grpc.ClientConn, projectID, topic string, opts *TopicOptions) (*pubsub.Topic, error)
```

Importantly, notice that the latter takes in a `*grpc.ClientConn`. Not doing so would be problematic.

Reply:
Ah ok. Updating...

Reply:
I would note though that for GCP, the Client already takes the projectID as a parameter, so that's no longer needed for the Topic constructor.